### PR TITLE
(docs): update README.md

### PR DESCRIPTION
**Pull request by vladdoster**
- correct pronoun agreement
- reduce verbiage
- increase readability
- correct capitalization
- make capitalization cohesive
- add punctuation
- make `md` section spacing cohesive
- make bolding of bullet items cohesive (i.e., **Note:** versus **Note**:)

Tangential comment, but I enjoyed learning about this project in the recent `Python Bytes` podcast  :^).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,44 +252,44 @@ Apache Airflow version life cycle:
 
 | Version | Current Patch/Minor | State     | First Release | Limited Support | EOL/Terminated |
 |---------|---------------------|-----------|---------------|-----------------|----------------|
-| 2       | 2.1.3               | Supported | Dec 17, 2020  | Dec 2021        | TBD            |
+| 2.0     | 2.1.3               | Supported | Dec 17, 2020  | Dec 31, 2021    | TBD            |
```
Comment (potiuk) [inline README.md]:
```suggestion
| 2       | 2.1.3               | Supported | Dec 17, 2020  | Dec 31, 2021    | TBD            |
```

This was correct. This line is about whole Airflow 2 line.

File: README.md [hunk 2]
```diff
@@ -254,44 +252,44 @@ Apache Airflow version life cycle:
...
 | 1.10    | 1.10.15             | EOL       | Aug 27, 2018  | Dec 17, 2020    | June 17, 2021  |
 | 1.9     | 1.9.0               | EOL       | Jan 03, 2018  | Aug 27, 2018    | Aug 27, 2018   |
 | 1.8     | 1.8.2               | EOL       | Mar 19, 2017  | Jan 03, 2018    | Jan 03, 2018   |
-| 1.7     | 1.7.1.2             | EOL       | Mar 28, 2016  | Mar 19, 2017    | Mar 19, 2017   |
+| 1.7     | 1.7.12              | EOL       | Mar 28, 2016  | Mar 19, 2017    | Mar 19, 2017   |
```
Comment (potiuk) [inline README.md]:
```suggestion
| 1.7     | 1.7.1.2             | EOL       | Mar 28, 2016  | Mar 19, 2017    | Mar 19, 2017   |
```

This was correct as well - see https://airflow.apache.org/announcements/#may-20-2016

Comment (eladkal) [inline README.md]: this is wrong. The version is 1.7.1.2 
https://airflow.apache.org/announcements/#may-20-2016
https://github.com/apache/airflow/releases/tag/1.7.1.2

File: README.md [hunk 3]
```diff
@@ -254,44 +252,44 @@ Apache Airflow version life cycle:
...
 3. We support a new version of Python/Kubernetes in main after they are officially released, as soon as we
    make them work in our CI pipeline (which might not be immediate due to dependencies catching up with
-   new versions of Python mostly) we release a new images/support in Airflow based on the working CI setup.
+   new versions of Python mostly) we release new images/support in Airflow based on the working CI setup.
 
 ### Additional notes on Python version requirements
 
-* Previous version [requires](https://github.com/apache/airflow/issues/8162) at least Python 3.5.3
-  when using Python 3
+* Previous versions [require](https://github.com/apache/airflow/issues/8162) at least Python 3.5.3
```
Comment (potiuk) [inline README.md:291]:
```suggestion
* Airflow 1.10 versions [require](https://github.com/apache/airflow/issues/8162) at least Python 3.5.3
```

**Review by potiuk — COMMENTED** [review]
Few NITS

File: README.md (hunk 1 above)
Comment (eladkal) [inline README.md]: The version series is 2 not 2.0
In Airflow 1 - a breaking change could be introduced between 1.x to 1.y which is why each version is in a separate row.
In Airflow 2 - we changed to [semver](https://airflow.apache.org/docs/apache-airflow/stable/release-process.html#airflow-s-release-process) so it doesn't require new row per patch version.

Also I'm not sure if the date for limited support is Dec 31. We don't know it yet.

File: README.md (hunk 3 above)
Comment (vladdoster) [reply README.md:291]: Your suggestion seems a bit ambiguous. I could interpret it as version 1.10 and newer require Python 3.5.3 or just version 1.10 requires Python 3.5.3+?

File: README.md (hunk 2 above)
Comment (vladdoster) [reply README.md]: Sorry for my assumption.

File: README.md (hunk 3 above)
Comment (potiuk) [reply README.md:291]: 1.10. Newer versions do not work with 3.5 at all

Comment (potiuk) [reply README.md:291]: actually we can remove the whole chapter I think

Comment (vladdoster) [reply README.md:291]: ok

**Review by potiuk — APPROVED** [review]

**Pull request merged**